providers/record.rb

```diff
@@ -22,7 +25,22 @@ def create_record(name, value, region, zone_id, record_type, ttl)
     )
   end
 
-  create_record(new_resource.name, new_resource.value, new_resource.region, new_resource.zone_id, new_resource.type, new_resource.ttl)
+  count = 0
+  while count <= record_wait_tries do
```

Comment:
Is there a ruby way to encapsulate this retry logic? I'm assuming there is since ruby tends to push DRY principle, I'm just not familiar enough with the techniques here. In python we'd just create a retry decorator.